packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -85,71 +91,63 @@ async function proxyPreviewModeApiRouteHandler(
     } catch {}
   }
 
-  previewModeProxy.once('proxyReq', proxyReq => {
-    proxyReq.removeHeader('X-Makeswift-Preview-Mode')
-
-    // The following headers are Next.js-specific and are removed to prevent Next.js from
-    // short-circuiting requests, breaking routing.
-    proxyReq.removeHeader('X-Invoke-Path')
-    proxyReq.removeHeader('X-Invoke-Query')
-
-    const url = new URL(proxyReq.path, 'http://n')
-
-    url.searchParams.delete('x-makeswift-preview-mode')
+  const {
+    'X-Makeswift-Preview-Mode': xMakeswiftPreviewMode,
+    'X-Invoke-Path': xInvokePath,
+    'X-Invoke-Query': xInvokeQuery,
+     ...headers
+  } = req.headers
 
-    proxyReq.path = url.pathname + url.search
-  })
-
-  if (req.query.secret !== apiKey) return res.status(401).send('Unauthorized')
+  const isForwardedProtoHttps = match(req.headers['X-Forwarded-Proto'])
```

Comment:
Doesn't `red.headers` keep headers in lowercase? If we want to avoid case-sensitivity we might want to use `getHeader`? Otherwise maybe we want to keep the behavior we had before?

Reply:
> Key-value pairs of header names and values. Header names are lower-cased.

Ahh yeah, it'll fix that to use lowercase, which I think we would prefer?

Reply:
Yeah, let's just keep the code as it was before.